Barotrauma/BarotraumaClient/ClientSource/Screens/SubEditorScreen.cs

```diff
@@ -2973,13 +2974,25 @@ private void FilterEntities(string filter)
 
             allEntityList.Visible = true;
             categorizedEntityList.Visible = false;
-            filter = filter.ToLower();
+
+            Regex r;
+            try
+            {
+                r = new Regex(filter, RegexOptions.IgnoreCase);
+            }
+            catch (ArgumentException)
+            {
+                r = null;
+            }
+
             foreach (GUIComponent child in allEntityList.Content.Children)
             {
+                bool shouldShow = r?.IsMatch(((MapEntityPrefab)child.UserData).Name) ?? false;
```

Comment:
As an additional thought, given that there is no feedback to the user if the regex fails to compile, it may be worth falling back to the string comparison in the case where `r` is `null` here.

Reply:
Was thinking about that. Added the old comparison back